### PR TITLE
Fix: A sequence_file with submitter file_source fails requirement check (#3202)

### PR DESCRIPTION
**Pull request by danielsotirhos**
…(#3202)

Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [ ] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [ ] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in sandbox and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [x] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [x] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [x] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [x] Verified that `N reviews` labelling is accurate
- [x] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [x] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [x] Started reindex in `dev`                              <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [x] Checked for failures in `dev`                         <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [x] Started reindex in `prod`                             <sub>or this PR does not require reindexing or does not target `prod`</sub>
- [x] Checked for failures in `prod`                        <sub>or this PR does not require reindexing or does not target `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

File: src/azul/plugins/metadata/hca/transform.py [hunk 1]
```diff
@@ -250,18 +250,8 @@ def from_index(self, value: str) -> Optional[JSON]:
...
-    """
-    # Note that currently both SubmitterCategory values allow the same set of
-    # entity types, so the order of types is swapped to provide unique values.
-    internal = api.SupplementaryFile, api.AnalysisFile
-    external = api.AnalysisFile, api.SupplementaryFile
-
-    def __init__(self, *file_types: Type[api.File]) -> None:
-        super().__init__()
-        self.file_types = file_types
+    internal = 1
```
Comment (jessebrennan) [inline src/azul/plugins/metadata/hca/transform.py:253]: Could use enum.auto().

**Review by jessebrennan — APPROVED** [review]
Looks good! Just one small suggestion.

File: src/azul/plugins/metadata/hca/transform.py [hunk 2]
```diff
@@ -250,18 +251,8 @@ def from_index(self, value: str) -> Optional[JSON]:
 
 
 class SubmitterCategory(Enum):
-    """
-    The types of submitters and the types of metadata entities that describe
-    the files they submit.
-    """
```
Comment (hannes-ucsc) [inline src/azul/plugins/metadata/hca/transform.py:256]: We need some sort of docstring.

File: src/azul/plugins/metadata/hca/transform.py [hunk 3]
```diff
@@ -1000,13 +990,11 @@ def _matrices(self, file: api.File) -> MutableJSON:
             # provided in the 'file_description' field of the file JSON.
             strata_string = file.json['file_description']
             matrix_cell_count = None
-        elif isinstance(file, api.AnalysisFile):
-            # Stratification values for analysis files are gathered by
+        else:
```
Comment (hannes-ucsc) [inline src/azul/plugins/metadata/hca/transform.py:993]:
```suggestion
        elif isinstance(file, api.File):
```

File: src/azul/plugins/metadata/hca/transform.py [hunk 4]
```diff
@@ -1000,13 +990,11 @@ def _matrices(self, file: api.File) -> MutableJSON:
...
+            # Stratification values for other file types are gathered by
             # visiting the file and using values from the graph.
             strata_string = self._build_strata_string(file)
             matrix_cell_count = file.matrix_cell_count
-        else:
-            assert False, type(file)
```
Comment (hannes-ucsc) [inline src/azul/plugins/metadata/hca/transform.py:1009]: Leave this in.

File: src/azul/plugins/metadata/hca/transform.py [hunk 5]
```diff
@@ -1000,13 +990,11 @@ def _matrices(self, file: api.File) -> MutableJSON:
             # provided in the 'file_description' field of the file JSON.
             strata_string = file.json['file_description']
             matrix_cell_count = None
-        elif isinstance(file, api.AnalysisFile):
-            # Stratification values for analysis files are gathered by
+        else:
+            # Stratification values for other file types are gathered by
```
Comment (hannes-ucsc) [inline src/azul/plugins/metadata/hca/transform.py:994]: Note that this doesn't match what you coded.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**